[roslaunch] Better exception handling when resource is not found.

130s opened the pull request:
When `$(find pkg)` fail to return a resource in `arg` tag, `roslaunch` fails without printing useful output. With this PR it provides better explanation.

Though I haven't looked into whether there's a test that covers this yet, I'm happy to add one if needed.

Without this PR:
```
$ roslaunch /tmp/invalid_arg.launch
:
Invalid <arg> tag: foo
ROS path [0]=/opt/ros/kinetic/share/ros
ROS path [1]=/home/n130s/ROS/indigo_trusty/cws_rosdt/src/ros/ros_comm/tools/roslaunch
ROS path [2]=/opt/ros/kinetic/share

Arg xml is <arg default="$(find foo)/.config" name="foo"/>
The traceback for the exception was written to the log file
```

With this PR:
```
$ roslaunch /tmp/invalid_arg.launch
:
Invalid <arg> tag: Make sure the following is found in ROS_PACKAGE_PATH: foo
ROS path [0]=/opt/ros/kinetic/share/ros
ROS path [1]=/home/n130s/ROS/indigo_trusty/cws_rosdt/src/ros/ros_comm/tools/roslaunch
ROS path [2]=/opt/ros/kinetic/share

Arg xml is <arg default="$(find foo)/.config" name="foo"/>
The traceback for the exception was written to the log file
```

```
$ more /tmp/invalid_arg.launch
<?xml version="1.0"?>
<launch>
  <arg name="foo" default="$(find foo)/.config" />
  <arg name="baa" default="$(arg foo)/hoge.yaml" />
</launch>
```

130s committed:
[roslaunch] Better exception handling when resource is not found.

When `$(find pkg)` fail to return a resource in `arg` tag, `roslaunch` fails without printing useful output. With this PR it provides better explanation.

Without this PR:
```
$ roslaunch /tmp/invalid_arg.launch
:
Invalid <arg> tag: foo
ROS path [0]=/opt/ros/kinetic/share/ros
ROS path [1]=/home/n130s/ROS/indigo_trusty/cws_rosdt/src/ros/ros_comm/tools/roslaunch
ROS path [2]=/opt/ros/kinetic/share

Arg xml is <arg default="$(find foo)/.config" name="foo"/>
The traceback for the exception was written to the log file
```

With this PR:
```
$ roslaunch /tmp/invalid_arg.launch
:
Invalid <arg> tag: Make sure the following is found in ROS_PACKAGE_PATH: foo
ROS path [0]=/opt/ros/kinetic/share/ros
ROS path [1]=/home/n130s/ROS/indigo_trusty/cws_rosdt/src/ros/ros_comm/tools/roslaunch
ROS path [2]=/opt/ros/kinetic/share

Arg xml is <arg default="$(find foo)/.config" name="foo"/>
The traceback for the exception was written to the log file
```

```
$ more /tmp/invalid_arg.launch
<?xml version="1.0"?>
<launch>
  <arg name="foo" default="$(find foo)/.config" />
  <arg name="baa" default="$(arg foo)/hoge.yaml" />
</launch>
```

dirk-thomas commented on tools/roslaunch/src/roslaunch/xmlloader.py:
```diff
@@ -301,6 +302,9 @@ def _arg_tag(self, tag, context, ros_config, verbose=True):
                 "arg '%s' is not defined. \n\nArg xml is %s"%(e, tag.toxml()))
+        except ResourceNotFound as e:
+            raise XmlParseException(
+                "Invalid <arg> tag: Make sure the following is found in ROS_PACKAGE_PATH: %s \n\nArg xml is %s"%(e, tag.toxml()))
```
In this case the tag itself is not "invalid". Also this is not a problem "parsing the XML" so the exception type is confusing.

130s replied:
That makes sense. Please see the update.

130s committed:
[roslaunch] Raise appropriate exception when a resource was not found.
Applyin feedback https://github.com/ros/ros_comm/pull/1476#discussion_r209081340

130s commented:
Output with the updated commit :
```
$ roslaunch /tmp/invalid_arg.launch
... logging to /home/n130s/.ros/log/0cae8616-9c9f-11e8-9714-c85b766a6139/roslaunch-130s-p50-23667.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

Resource not found: Tag the exception occurred at <arg default="$(find foo)/.config" name="foo"/>
Package not found in ROS_PACKAGE_PATH: foo
ROS path [0]=/opt/ros/kinetic/share/ros
ROS path [1]=/home/n130s/cws_rosdt/install/share
ROS path [2]=/home/n130s/cws_rosdt/src/ros/ros_comm/tools/roslaunch
ROS path [3]=/opt/ros/kinetic/share
Make sure the package is found in ROS_PACKAGE_PATH.
The traceback for the exception was written to the log file
```

130s commented:
@ros-pull-request-builder retest this please

dirk-thomas committed:
shortened the exception message a bit

dirk-thomas commented:
@130s I hope the shorter message is fine with you: 9b3f5e37ca656bb9659d3694e53879dd4fb220a8

dirk-thomas commented:
Thank you for the improvement.
